lib/generateTasks.js

```diff
@@ -16,43 +29,57 @@ const debug = require('debug')('lint-staged:gen-tasks')
  * @param {boolean} [options.files] - Staged filepaths
  * @param {boolean} [options.relative] - Whether filepaths to should be relative to gitDir
  */
-module.exports = function generateTasks({
-  config,
-  cwd = process.cwd(),
-  gitDir,
-  files,
-  relative = false,
-}) {
+const generateTasks = (
+  { config, cwd = process.cwd(), gitDir, files, relative = false },
+  logger
+) => {
   debug('Generating linter tasks')
 
   const absoluteFiles = files.map((file) => normalize(path.resolve(gitDir, file)))
   const relativeFiles = absoluteFiles.map((file) => normalize(path.relative(cwd, file)))
 
-  return Object.entries(config).map(([pattern, commands]) => {
+  return Object.entries(config).map(([rawPattern, commands]) => {
+    let pattern = rawPattern
+
     const isParentDirPattern = pattern.startsWith('../')
 
-    const fileList = micromatch(
-      relativeFiles
-        // Only worry about children of the CWD unless the pattern explicitly
-        // specifies that it concerns a parent directory.
-        .filter((file) => {
-          if (isParentDirPattern) return true
-          return !file.startsWith('..') && !path.isAbsolute(file)
-        }),
-      pattern,
-      {
-        cwd,
-        dot: true,
-        // If pattern doesn't look like a path, enable `matchBase` to
-        // match against filenames in every directory. This makes `*.js`
-        // match both `test.js` and `subdirectory/test.js`.
-        matchBase: !pattern.includes('/'),
-      }
-    ).map((file) => normalize(relative ? file : path.resolve(cwd, file)))
+    // Only worry about children of the CWD unless the pattern explicitly
+    // specifies that it concerns a parent directory.
+    const filteredFiles = relativeFiles.filter((file) => {
```

Comment:
Not a real issue, but I have a habit to always provide a bit of advice if I can:
 `generateTask.js` is more about "generate" and "task", and this moment is more about clearing the user input and some sort of validation. 
 Wondering - is it possible to move this logic into `validateConfig`, where it _gravitates_ more remove connectivity with `messages` and `logger`. However, it will also create new connectivity between `generateTasks` and `validateConfig` which does not exist currently.

Reply:
Yes, I actually had the same thought in https://github.com/okonet/lint-staged/pull/992#issuecomment-879134808 after making these changes.

It makes sense to move it there, however currently `validateConfig` mostly just throws errors, and in this case it would simply warn. I guess that's ok though.

Reply:
This should now be moved to `validateConfig`